pkg/handlers/v1/filter.go

```diff
@@ -0,0 +1,34 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/asecurityteam/nexpose-vuln-filter/pkg/domain"
+	"github.com/asecurityteam/nexpose-vuln-filter/pkg/domain/nexpose"
+)
+
+// NexposeAssetVulnerabilities is a Nexpose asset response payload appended
+// with assetVulnerabilityDetails
+type NexposeAssetVulnerabilities struct {
+	nexpose.Asset
+	Vulnerabilities []nexpose.AssetVulnerabilityDetails `json:"assetVulnerabilityDetails"`
+}
+
+// NexposeVulnFilter accepts a payload with Nexpose asset information
+// and a list of vulnerabilities and returns a payload of the same shape
+// omitting vulnerabilities that do not meet the filter criteria
+type NexposeVulnFilter struct {
+	VulnerabilityFilter domain.VulnFilterer
+	LogFn               domain.LogFn
+}
+
+// Handle filters an asset's vulnerabilities based on predefined criteria and returns
+// the payload without vulnerabilities that meet the requirements
+func (h NexposeVulnFilter) Handle(ctx context.Context, input *NexposeAssetVulnerabilities) (*NexposeAssetVulnerabilities, error) {
+	logger := h.LogFn(ctx)
+	initialNumberOfVulns := len(input.Vulnerabilities)
+	input.Vulnerabilities = h.VulnerabilityFilter.FilterVulnerabilities(input.Vulnerabilities)
```

Comment:
Here you are modifying the input parameter. This isn't necessarily a bad thing to do, but I would exercise caution since the invocation path is owned by a different project.  This has the potential to lead to unknown consequences later.